lib/es6/caml.js

```diff
@@ -39,6 +39,20 @@ function float_compare(x, y) {
   }
 }
 
+function bigint_compare(x, y) {
+  if (x === y) {
+    return 0;
+  } else if (x < y) {
+    return -1;
+  } else if (x > y || x === x) {
```

Comment:
Weird or clause

Reply:
Oh good catch, I didn't notice because I was using the float_compare code as is. Why does float_compare have this branching as opposed to int_compare? Very weird to me too.

Reply:
I think those conditions are for below test cases;
```res
// float_test.res
eq(__LOC__, float_compare(Js.Float._NaN, Js.Float._NaN), 0)
eq(__LOC__, generic_compare(Js.Float._NaN, Js.Float._NaN), 0)
eq(__LOC__, float_compare(Js.Float._NaN, neg_infinity), -1)
eq(__LOC__, generic_compare(Js.Float._NaN, neg_infinity), -1)
```

Reply:
I don't know why float_compare was implemented this way at the time, but my best guess is that it was to capitalize on the fact that comparing NaNs is false.
```js
-Infinity === -Infinity // true
NaN === NaN // false
```
However, the bigint doesn't have a binding to Infinity or Nan for now. I'm going to fix `bigint_compare` as `int_compare`.

Reply:
생각을 바꿨습니다. `Js.Bigint`에도 `NaN`과 method 들에 대한 interop을 추가하고,`bigint_test.res` 테스트도 추가해보았습니다. 한 번 봐주시겠어요? https://github.com/rescript-lang/rescript-compiler/pull/6670/commits/c972a1374643aaf566b544c41c89ba5fa02c146e

Reply:
I changed my mind. I've added interop for `NaN` and methods to Js.Bigint, and also added the `bigint_test.res` test. Can you take a look? https://github.com/rescript-lang/rescript-compiler/commit/c972a1374643aaf566b544c41c89ba5fa02c146e

Reply:
I can not say that the implementation of `float_compare` is the best but it is practical and working.

Reply:
NaN is a concept that belongs to Number (`typeof NaN === `number`), not BigInt. BigInt operations never produce NaN AFAIK

Reply:
That is the compare function. Those clauses are for the comparing with NaN.

Reply:
I got your point. NaN can not be produced with BigInt operation. NaN is produced only below 5 cases:
> There are five different types of operations that return NaN:
>
> * Failed number conversion (e.g. explicit ones like parseInt("blabla"), Number(undefined), or implicit ones like Math.abs(undefined))
> * Math operation where the result is not a real number (e.g. Math.sqrt(-1))
> * Indeterminate form (e.g. 0 * Infinity, 1 ** Infinity, Infinity / Infinity, Infinity - Infinity)
> * A method or expression whose operand is or gets coerced to NaN (e.g. 7 ** NaN, 7 * "blabla") — this means NaN is contagious
> * Other cases where an invalid value is to be represented as a number (e.g. an invalid [Date](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) new Date("blabla").getTime(), "".charCodeAt(1))

Reply:
So wouldn't it be better to get rid of NaNs in Js_bigint as well?

Reply:
Yes. NaN is obviously number type. Date, parseInt, Math are all about Number, not BigInt.

Reply:
Thanks!

https://github.com/rescript-lang/rescript-compiler/pull/6670/commits/6601152d3b720a7fdf162c2c22807d5a2ced29ac
https://github.com/rescript-lang/rescript-compiler/pull/6670/commits/ea49c0534c92ca4e552c8fbca1c9d533d44a4504